peldd.cc

```diff
@@ -417,13 +413,13 @@ string Path_Cache::resolve(const deque<string> &search_path,
         if (i == m_.end()) {
           unordered_map<string, string> xs;
           for (auto &e : fs::directory_iterator(path)) {
-            auto fn = e.path().filename().generic_string();
+            auto fn = e.path().filename().string();
             xs[_to_lower(fn)] = std::move(fn);
           }
           auto r = m_.insert(make_pair(path, std::move(xs)));
-          return path + "/" + resolve(r.first->second, filename);
+          return path + (char)fs::path::preferred_separator + resolve(r.first->second, filename);
```

Comment:
`std::filesystem::path`, and the equivalent boost things, have an overloaded `operator/` that inserts the preferred directory separator.

So you can just do `return path / resolve(r.first->second, filename);` or, at worst, `fs::path{path} / resolve(r.first->second, filename);`

Reply:
https://en.cppreference.com/w/cpp/filesystem/path/operator_slash

Reply:
Using `operator/` would not only require explicitly casting one of the operators to `fs::path` as in your second example, it would also require us to convert the path back to string with the `string()` method. So it would need to be `return (fs::path{path} / resolve(r.first->second, filename)).string()`, which seems much harder to read in my opinion.